modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/product_detail/render/view.jsp

```diff
@@ -73,22 +73,60 @@ long cpDefinitionId = cpCatalogEntry.getCPDefinitionId();
 					<aui:button cssClass="btn btn-primary btn-sm my-2" href="<%= cpContentHelper.getReplacementCommerceProductFriendlyURL(cpSku, themeDisplay) %>" value="replacement-product" />
 				</c:if>
 
-				<c:if test="<%= (cpSku != null) && (cpSku.getDiscontinuedDate() != null) %>">
-
-					<%
-					Format format = FastDateFormatFactoryUtil.getSimpleDateFormat("MMMMM dd, yyyy", locale, timeZone);
-					%>
+				<%
+				Format dateFormat = FastDateFormatFactoryUtil.getSimpleDateFormat("MMMMM dd, yyyy", locale, timeZone);
+				%>
 
+				<c:if test="<%= (cpSku != null) && (cpSku.getDiscontinuedDate() != null) %>">
 					<p class="my-2">
 						<span class="font-weight-semi-bold">
 							<%= LanguageUtil.get(request, "end-of-life") %>
 						</span>
 						<span>
-							<%= format.format(cpSku.getDiscontinuedDate()) %>
+							<%= dateFormat.format(cpSku.getDiscontinuedDate()) %>
```

Comment:
It's ok to format the date, but what happens if I select another SKU? Is this info still there?

Reply:
For multiple sku products there is a problem because, the jsp set cpSku to  cpContentHelper.getDefaultCPSku(cpCatalogEntry), and with multiple sku products is null. Because it's null it's not showed. But it's a behavior that was already in this page

Reply:
But what if the other SKU has a Discontinued date? Would it be shown or not? I doubt it from the code.

Reply:
no it's showed, but it's a behavior that was already in that page for that javascipt ajax thing